server/lib/chat/chat-api.ts

```diff
@@ -337,21 +342,18 @@ export class ChatApi {
   }
 
   @httpGet('/:channelId(\\d+)')
-  @httpBefore(throttleMiddleware(retrievalThrottle, ctx => String(ctx.session!.userId)))
+  @httpBefore(throttleMiddleware(channelRetrievalThrottle, ctx => String(ctx.session!.userId)))
   async getChannelInfo(ctx: RouterContext): Promise<GetChannelInfoResponse> {
     const channelId = getValidatedChannelId(ctx)
 
     return await this.chatService.getChannelInfo(channelId, ctx.session!.userId)
   }
-}
-
-@httpApi('/admin/chat')
-@httpBeforeAll(ensureLoggedIn, convertChatServiceErrors)
-export class AdminChatApi {
-  constructor(private chatService: ChatService) {}
 
   @httpGet('/')
-  @httpBefore(checkAllPermissions('moderateChatChannels'))
+  @httpBefore(
+    featureEnabled(MULTI_CHANNEL),
+    throttleMiddleware(channelRetrievalThrottle, ctx => String(ctx.session!.userId)),
+  )
   async searchChannels(ctx: RouterContext): Promise<SearchChannelsResponse> {
     const {
       query: { q: searchQuery, limit, page },
```

Comment:
`limit` should almost certainly have a maximum

Reply:
`page` just seems weird as hell too, why are we passing in a value that we have to multiply by another value to get to anything useful (and the client likely has to divide to send to us)? Just pass the offset

Reply:
> `page` just seems weird as hell too, why are we passing in a value that we have to multiply by another value to get to anything useful (and the client likely has to divide to send to us)? Just pass the offset

Hmm, the client just increases the `page` by 1 when it reaches the end of a list and wants to load more channels. But I guess increasing the current offset by the `limit` works too.

Reply:
You don't have to do any math, the list already contains the number of elements you have seen and is thus the starting point of the next request. The only time you need to deal with the limit is to know if you received less than you asked for (so you're done), but that should probably be a `boolean` returned in the response anyway